### ISSUE TITLE
LongMethod counts statements but not actual lines as its description says

### ISSUE THREAD

**Issue by dimsuz** [arturbosch/detekt#1279]
The rule or changes you're looking for might have already been suggested!
Please search in the [issues](https://github.com/arturbosch/detekt/issues) before creating a new one.

## Expected Behavior of the rule
LongMethod should count actual line breaks instead of statements

## Context
I have the following draft method (ignore its complexity, it's due to the fact that I'm drafting it :smile_cat:)

```kotlin
private fun Response.PeriodKindParams.extractValue(): PeriodKindValue {
  val itemsSafe = this.items.requireField("periodKind items")
  val key = defaultValue
    ?: itemsSafe.entries.firstOrNull()?.key
    ?: throw DataMappingException("periodKind items list is empty")
  val title = itemsSafe[key] ?: throw NoSuchElementException("no key=$key in items list")
  return when (key.toLowerCase()) {
    "month" -> {
      PeriodKindValue.MonthDay(
        period = LookUpValue(LookUpItem(key, title)),
        periodValue = IntegerValue(
          minValue = month?.minValue ?: Int.MIN_VALUE,
          maxValue = month?.maxValue ?: Int.MAX_VALUE,
          currentValue = month?.defaultValue ?: 0
        )
      )
    }
    "week" -> {
      val weekItems = week?.items.requireField("week items")
      val weekDefaultKey = week?.defaultValue
        ?: weekItems.entries.firstOrNull()?.key
        ?: throw DataMappingException("periodKind week items is empty")
      val weekDefaultValue = weekItems[weekDefaultKey]
        ?: throw NoSuchElementException("no key=$weekDefaultKey in week item list")
      val item = LookUpItem(weekDefaultKey, weekDefaultValue)

      PeriodKindValue.WeekDay(
        period = LookUpValue(LookUpItem(key, title)),
        periodValue = LookUpValue(item)
      )
    }
    else -> PeriodKindValue.Other(LookUpValue(LookUpItem(key, title)))
  }
}
```
It's obvious that it is large and should be refactored into a smaller functions.
I was very surprised that the only thing `detekt` complained about was `ThrowsCount` for this method.

I found that `LongMethod` was counting statements which I suggest to change. 
If you agree I might prepare a PR once I have time.

**Comment by dimsuz** [comment]
Do you think it should skip lines which contain only `\n` or trailing whitespace?
Or it should be made an configurable option? I'm not sure here.

**Comment by arturbosch** [comment]
Yep, blank lines should be skipped. Normally there is a distinction between LOC = #lines, SLOC = lines of source code, CLOC = comment lines of code, just blanks.
I would expect to count SLOC. And this brings the question what to do if the line starts with a comment but also has some code later? 
I think the best way would be to use a scanner and see token for token what it is and calculate the loc based on the lines of the token.

**Comment by dimsuz** [comment]
Oh wait, there's already [SLOCVisitor](https://github.com/arturbosch/detekt/blob/eaa4de29b8a22e23dcfed15baae137147d3b563d/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/processors/ProjectSLOCProcessor.kt#L13) in the project, but it simply counts number of lines in file, while I want to create something more intelligent which would count SLOC in `KtNamedFunction` or whatever construct - based on whitespace tokens (`PsiWhitespace`). 

The plan is to override `visitWhitespace` and then work from there. I guess I could either extend this exisiting class or create some new one. It seems like this implementation is internal to `ProjectSLOCProcessor` though, but I'm not sure if this is true.

**Comment by arturbosch** [comment]
There is also the `internal` package in the api module. If you can rework the SLOCVisitor to support all cases it would be awesome, thanks! :)

**Comment by arturbosch** [comment]
Hey, I've tried out to just count source tokens and it resulted in a fix for this in #1448.

**Comment by dimsuz** [comment]
Oh, great! No, I didn't have a chance to continue work on this, so this is nice you solved it! Looking forward to seeing this in a new release.
